### PR TITLE
#43 #55 Search Phone contacts to invite + See list of people in my monitoring group

### PR DESCRIPTION
**Pull request by KennanHoa**
#43 
#55 

- Clicking on Monitoring Group should show the list of users in the group you are monitoring. Search function should be working.
- Clicking on "Phone Contacts" should show list of people from the phone local Contacts app with their phone number under the name.

**For Contacts, please go to the BigOwlApp.apk settings and allow permission to use Contacts. (Only need to do once)**

### CODE REVIEW

**Comment by KhangZhengNg** [comment]
Do git update-index --chmod=+x gradlew on your terminal to fix the error "./gradlew: Permission denied" on github-actions

File: app/src/main/java/com/example/bigowlapp/HomeActivity.java [hunk 1]
```diff
@@ -61,7 +62,7 @@ public void onClick(View v) {
             btnMonitoringGroup.setOnClickListener(new View.OnClickListener() {
                 @Override
                 public void onClick(View v) {
-                    Intent i = new Intent(HomeActivity.this, MonitoringGroupPageActivity.class);
+                    Intent i = new Intent(HomeActivity.this, MonitoringGroupListActivity.class);
```
Comment (KhangZhengNg) [inline app/src/main/java/com/example/bigowlapp/HomeActivity.java:65]: Change the intent back to MonitoringGroupPageActivity.class as we only allowed 1 monitoring group per user for now. Keep the MonitoringGroupListActivity file for future possible feature.

**Comment by ShawnGuerra-Bautista** [comment]
image

When I click on the Monitoring group button, the app crashes.

File: app/src/main/java/com/example/bigowlapp/SearchContactsToSupervise.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+                loadContacts();
+            //}
+        //});
+
+    }
+
+    private void loadContacts() {
+        //StringBuilder builder = new StringBuilder();
+        ContentResolver contentResolver = getContentResolver();
+        Cursor cursor = contentResolver.query(ContactsContract.Contacts.CONTENT_URI, null,null,null,null);
```
Comment (KhangZhengNg) [inline app/src/main/java/com/example/bigowlapp/SearchContactsToSupervise.java:58]: Got a runtime exceotion "requires android.permission.READ_CONTACTS or android.permission.WRITE_CONTACTS". Pop up the system alert window for the app to gain permission from user.

File: app/src/main/res/layout/activity_home.xml [hunk 3]
```diff
@@ -1,10 +1,10 @@
...
-    xmlns:app="http://schemas.android.com/apk/res-auto"
-    xmlns:tools="http://schemas.android.com/tools"
-    android:layout_width="match_parent"
-    android:layout_height="match_parent"
-    tools:context=".HomeActivity">
+xmlns:app="http://schemas.android.com/apk/res-auto"
+xmlns:tools="http://schemas.android.com/tools"
+android:layout_width="match_parent"
+android:layout_height="match_parent"
+tools:context=".HomeActivity">
```
Comment (KhangZhengNg) [inline app/src/main/res/layout/activity_home.xml:7]: Check for linting and spacing. do "Control+Alt+L" on android studio to do a Reformat code

File: app/src/main/java/com/example/bigowlapp/MonitoringGroupListActivity.java [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+        }*/
+        List<User> filteredUsers = mUsers.stream().filter(u -> {
+            boolean containInFirstName = u.getFirstName().toLowerCase().contains(s);
+            boolean containInLastName = u.getLastName().toLowerCase().contains(s);
+            return (containInFirstName || containInLastName);
+        }).collect(Collectors.toList());
+
+        mUsersShow = filteredUsers;
+    }
+}
```
Comment (KhangZhengNg) [inline app/src/main/java/com/example/bigowlapp/MonitoringGroupListActivity.java:149]: As mentioned, keep this file for future possible feature

**Review by KhangZhengNg — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/bigowlapp/MonitoringGroupListActivity.java [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+                                                    supervisorName = findViewById(R.id.textView_supervisorName);
+                                                    supervisorName.setText(u.toString());
+                                                }
+                                            }
+                                        });
+                                    }
+
+                                    //List of all user to compare with
+                                    db.collection("users")
+                                            .whereIn(FieldPath.documentId(), mSupervisedGroup)
```
Comment (ShawnGuerra-Bautista) [inline app/src/main/java/com/example/bigowlapp/MonitoringGroupListActivity.java:88]: image

This relates to the issue when a new user clicks on the monitoring group button.

**Review by ShawnGuerra-Bautista — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/example/bigowlapp/SearchContactsToSupervise.java (hunk 2 above)
Comment (KennanHoa) [reply app/src/main/java/com/example/bigowlapp/SearchContactsToSupervise.java:58]: fixed

Comment (KennanHoa) [reply app/src/main/java/com/example/bigowlapp/SearchContactsToSupervise.java:58]: fixed

File: app/src/main/java/com/example/bigowlapp/Adapter/UserAdapter.java [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+import androidx.annotation.NonNull;
+
+import com.example.bigowlapp.model.User;
+import com.example.bigowlapp.R;
+
+
+import java.util.List;
+
+public class UserAdapter extends RecyclerView.Adapter<UserAdapter.ViewHolder> {
```
Comment (KhangZhengNg) [inline app/src/main/java/com/example/bigowlapp/Adapter/UserAdapter.java:18]: Good job on creating your own adapter for User!

File: app/src/main/java/com/example/bigowlapp/HomeActivity.java [hunk 7]
```diff
@@ -75,9 +85,63 @@ public void onClick(View v) {
...
         }
         catch (Exception ex)
         {
+
+        }
+    }
+
+    //TODO refactor to another class file
+    private void requestContactPermission(){
+        if (ActivityCompat.shouldShowRequestPermissionRationale(this, Manifest.permission.READ_CONTACTS)){
```
Comment (KhangZhengNg) [inline app/src/main/java/com/example/bigowlapp/HomeActivity.java:112]: Well done on asking for device permission from user!

**Review by KhangZhengNg — APPROVED** [review]

**Review by ShawnGuerra-Bautista — APPROVED** [review]

**Pull request merged**